sublist3r.py

```diff
@@ -915,8 +916,32 @@ def main(domain, threads, savefile, ports, silent, verbose, enable_bruteforce):
     if verbose and not silent:
         print(Y+"[-] verbosity is enabled, will show the subdomains results in realtime"+W)
 
+    supported_engines = {'baidu':BaiduEnum,
+                         'yahoo':YahooEnum,
+                         'google':GoogleEnum,
+                         'bing':BingEnum,
+                         'ask':AskEnum,
+                         'netcraft':NetcraftEnum,
+                         'dnsdumpster':DNSdumpster,
+                         'virstotal':Virustotal,
+                         'threatcrowd':ThreatCrowd,
+                         'ssl':CrtSearch,
+                         'passivedns':PassiveDNS
+                         }
+
+    chosenEnums = []
+
+    if engines == None:
+        chosenEnums = [BaiduEnum, YahooEnum, GoogleEnum, BingEnum, AskEnum,
+                       NetcraftEnum, DNSdumpster, Virustotal, ThreatCrowd, CrtSearch, PassiveDNS]
+    else:
+        engines = engines.split(',')
+        for engine in engines:
+            if supported_engines.has_key(engine.lower()):
```

Comment:
@aboul3la This breaks Python3 compatibility because 'has_key' function was removed in Python 3.